doc/topics/topology/syndic.rst

```diff
@@ -4,31 +4,46 @@
 Salt Syndic
 ===========
 
+.. warning::
+
+    Salt Syndic with :ref:`external auth or publisher_acl<acl-eauth>`
+    should not be used for security purposes.
+
+    The long-term goal for Syndics is that they work correctly with
+    authentication and authorization, but with Salt's current architecture that
+    Syndics work with publisher_acl/external auth at all should be considered a
+    convenience. Syndics with external auth or publisher_acl should not used
+    anywhere that security is considered essential. With the changes resulting
+    from `issue #62618 on GitHub
+    <https://github.com/saltstack/salt/issues/62618>`_, there have been some
+    improvements, but currently Syndic functionality is only fully supported
+    when running ``salt`` on the Master of Masters as the root/service user.
+
+    Additionally, it's technically possible to have minions of the same
+    name on different Syndics. In that case, Salt's behavior is completely
+    undefined and subject to change.
+
+    It may also be possible for a Syndic to itself be a Master of Masters, but
+    that behavior is also undocumented and should not yet be considered a
+    supported feature.
```

Comment:
@naaronne Moving & updating the below warning - Anil wanted to get your eyes on this since we're explicitly calling out that Syndics, despite their inclusion in Salt for several years, should not (yet) be treated as a security feature.

Before, we simply lacked any documentation about whether or not, or how Syndics might interact with publisher_acl or eauth.

This PR *does* restore the previous functionality, re-enabling eauth, but until we can take the time to really refactor Syndics, I would like to warn against relying on it for security.

Reply:
This documentation needs to be rewritten, it doesn't reflect the confidence of this fix.

Reply:
FWIW, this documentation is not documenting the fix as much as the general state of Syndics.

An analogy I thought of is that Syndics could be considered a chainsaw. There are plenty of people who use chainsaws on a daily basis, to no ill-effects, because they understand the dangers, wear the proper protective equipment, and have a safety attitude (e.g. don't operate while mentally/chemically impaired).

However, if you're not wearing the correct equipment, and you operate a chainsaw dangerously, you might take your leg off.

Just because I replaced the broken chain, doesn't mean that the chainsaw ceases to be dangerous. And putting up a new warning sign about the dangers of the chainsaw doesn't change anything about the chainsaw being a chainsaw. And it shouldn't imply that we're not confident about the chain. 

But I wouldn't go around juggling the chainsaw :slightly_smiling_face: